src/natcap/invest/habitat_quality.py

```diff
@@ -188,281 +191,326 @@
-                threat_band = threat_raster.GetRasterBand(1)
-                for block_offset in pygeoprocessing.iterblocks(
-                        (aligned_threat_path, 1), offset_only=True):
-                    block = threat_band.ReadAsArray(**block_offset)
-
-                    # First check if we actually need to set anything.
-                    # No need to perform unnecessary writes!
-                    if set(numpy.unique(block)) == set([0, 1]):
-                        continue
-
-                    zero_threat = numpy.isclose(block, threat_nodata)
-                    block[zero_threat] = 0
-                    block[~numpy.isclose(block, 0)] = 1
-
-                    threat_band.WriteArray(
-                        block, yoff=block_offset['yoff'],
-                        xoff=block_offset['xoff'])
-                threat_band = None
-                threat_raster = None
+                    aligned_bound_threat_path)
+
+                # Bound threat raster values to 0 <= x <= 1
+                bound_threat_values_task = task_graph.add_task(
+                    _bound_raster_values,
+                    args=(
+                        (aligned_threat_path, 1), aligned_bound_threat_path),
+                    target_path_list=[aligned_bound_threat_path],
+                    dependent_task_list=[align_task],
+                    task_name=f'bound_threat{lulc_key}_{threat}')
+                bound_threat_values_tasks.append(bound_threat_values_task)
 
     LOGGER.info('Starting habitat_quality biophysical calculations')
-
     # Rasterize access vector, if value is null set to 1 (fully accessible),
     # else set to the value according to the ACCESS attribute
     cur_lulc_path = lulc_path_dict['_c']
     fill_value = 1.0
-    try:
-        LOGGER.info('Handling Access Shape')
-        access_raster_path = os.path.join(
-            inter_dir, 'access_layer%s.tif' % suffix)
-        # create a new raster based on the raster info of current land cover
-        pygeoprocessing.new_raster_from_base(
-            cur_lulc_path, access_raster_path, gdal.GDT_Float32,
-            [_OUT_NODATA], fill_value_list=[fill_value])
-        pygeoprocessing.rasterize(
-            args['access_vector_path'], access_raster_path, burn_values=None,
-            option_list=['ATTRIBUTE=ACCESS'])
-
-    except KeyError:
-        LOGGER.info(
-            'No Access Shape Provided, access raster filled with 1s.')
+
+    LOGGER.info('Handling Access Shape')
+    access_raster_path = os.path.join(
+        intermediate_output_dir, f'access_layer{file_suffix}.tif')
+    # create a new raster based on the raster info of current land cover
+    create_access_raster_task = task_graph.add_task(
+        pygeoprocessing.new_raster_from_base,
+        args=(cur_lulc_path, access_raster_path, gdal.GDT_Float32,
+              [_OUT_NODATA]),
+        kwargs={
+            'fill_value_list': [fill_value]
```

Comment:
Is `fill_value` used anywhere else? If not, I think it's reasonable to drop it and hard-code a 1.0 here with a comment saying "we default access to 1.0 and change it when it's not true" or whatever makes sense.